File: lab07/log_probability.py
#!/usr/bin/python
import re, sys, glob,math
reg= r"\b"+re.escape(sys.argv[1])+r"\b"
for file in sorted(glob.glob("lyrics/*.txt")):
        count=0
        total=0
        content = open(file,"r")
        name = re.sub("lyrics/","",file)
        name = re.sub(".txt","",name)
        name = re.sub("_"," ",name)
        for line in content:
                line = line.lower()
                all = re.findall("[A-Za-z]+",line)
                array = re.findall(reg,line)
                total += len(all)
                count += len(array)
        value = math.log(float(count+1)/float(total))
        print("log((%d+1)/%6d) = %8.4f %s" %(count, total, value, name))

File: lab07/log_probability.pl
#!/usr/bin/perl
foreach $file (glob "lyrics/*.txt") {
        my $total=0;
        my $occur=0;
        open FILE ,'<',$file or die;
        @files=<FILE>;
        close FILE;
        $file =~ s/lyrics\///g;
        $file =~ s/.txt//g;
        $file =~ s/_/ /g;
        foreach $line (@files) {
                $line = lc $line;
                @all = $line =~ /[A-Za-z]+/g;
		$ARGV[0] = lc $ARGV[0];
                @array = $line =~ /\b$ARGV[0]\b/g;   #match a word boundary
                $total += $#all +1;
                $occur += $#array+1;
        }
        my $result=log(($occur+1)/$total);
        printf("log((%d+1)/%6d) = %8.4f %s\n","$occur","$total","$result","$file");
}


